Clockwise/TimeBudgetExtensions.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Threading.Tasks;
+
+namespace Clockwise
+{
+    public static class TimeBudgetExtensions
+    {
+        public static async Task CancelIfExceeds(
+            this Task task,
+            TimeBudget budget,
+            Action ifCancelled = null)
+            =>
+                await task.CancelAfter(budget.CancellationToken,
+                                       ifCancelled: () =>
+                                       {
+                                           if (ifCancelled == null)
+                                           {
+                                               throw new TimeBudgetExceededException(budget);
+                                           }
+                                           else
```

Comment:
this else is not strictly needed

Reply:
It's true. I was splitting it out to better see which lines of code are covered and pushing to see if a test issue would repro in CI. (It does.)